pkg/template/event.go

```diff
@@ -97,11 +102,22 @@ func applyEventValuesToParams(params []triggersv1.Param, body []byte, header htt
 		for i, expr := range expressions {
 			val, err := ParseJSONPath(event, expr)
 			if err != nil {
-				return nil, fmt.Errorf("failed to replace JSONPath value for param %s: %s: %w", p.Name, p.Value, err)
+				return nil, fmt.Errorf(jsonPathErr+"%s: %s: %w", p.Name, p.Value, err)
 			}
 			pValue = strings.ReplaceAll(pValue, originals[i], val)
 		}
 		params[idx].Value = pValue
 	}
 	return params, nil
 }
+
+// IsParseErr determines if err is an error which indicates that parsing event failed
+func IsParseErr(err error) bool {
+	if strings.Contains(err.Error(), jsonPathErr) {
```

Comment:
I don't think we should determine error type based on the strings (e.g. could I embed extra data in interceptor responses that would cause this to pass?)

If we go with this style of error handling, we should likely look into error wrapping instead. https://blog.golang.org/go1.13-errors

Reply:
This would also allow us to not tie BadRequest statuses to marshal errors, since we could infer it from the error type instead.